File: solutions/c/armstrong-numbers/1/armstrong_numbers.c
#include "armstrong_numbers.h"

int _pow(int num, int e)
{
    int i;
    int result = 1;
    for (i = 0; i < e; ++i) {
        result *= num;
    }

    return result;
}

bool is_armstrong_number(int candidate)
{
    int sum = 0;
    int digits = 0;
    int copy = candidate;
    while (copy > 0) {
        digits++;
        copy /= 10;
    }

    copy = candidate;
    while (copy > 0) {
        sum += _pow((copy % 10), digits);
        copy /= 10;
    }

    return candidate == sum;
}

File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
final class ArmstrongNumbers {

    public boolean isArmstrongNumber(final int numberToCheck) {
        int number = numberToCheck;
        int digitCount = 0;
        while (number > 0) {
            number /= 10;
            digitCount++;
        }

        long maybeArmstrongNumber = 0;
        number = numberToCheck;
        while (number > 0) {
            final int digit = number % 10;
            number /= 10;
            maybeArmstrongNumber += (long) Math.pow(digit, digitCount);
        }

        return maybeArmstrongNumber == numberToCheck;
    }
}
